src/NHibernate/Impl/SessionImpl.cs

```diff
@@ -140,15 +126,19 @@ void ISerializable.GetObjectData(SerializationInfo info, StreamingContext contex
 			{
 				throw new InvalidOperationException("Cannot serialize a Session while connected");
 			}
+			if (_transactionCoordinatorShared)
+			{
+				throw new InvalidOperationException("Cannot serialize a Session sharing its transaction coordinator");
```

Comment:
After deserialization, the shared thing (currently connection manager) would be no more shared.